jerry-core/include/jerryscript-core.h

```diff
@@ -159,6 +159,18 @@ typedef struct
 } jerry_property_descriptor_t;
 
 /**
+ * Description of JerryScript heap memory stats.
+ * It is for memory profiling.
+ */
+typedef struct
+{
+  size_t version; /**< the version of the stats struct */
```

Comment:
> The problem is if you dynamically link Jerry, and you compiled your code with an older Jerry, and the system has a newer one, you will experience a crash.

In the current implementation doesn't know or check the version / how large the caller's `jerry_heap_stats_t` struct is. I think the caller should be passing in the desired/expected version then for this to work?